app/data_providers/curation_api_client.py

```diff
@@ -0,0 +1,48 @@
+import uuid
+from abc import ABC, abstractmethod
+from typing import List
+
+import requests
+import json
+from datetime import date
+
+from app.graphql.corpus_item import CorpusItem
+from app.models.corpus_item_model import CorpusItemModel
+from app.models.ranked_corpus_items_instance import RankedCorpusItemsInstance
+
+class CurationAPIFetchable(ABC):
+    @abstractmethod
+    async def get_ranked_corpus_slate(self, corpus_id: str, start_date: str, user_id) -> List[CorpusItem]:
+        return NotImplemented
+
+class CurationAPIClient(CurationAPIFetchable):
+    async def get_ranked_corpus_slate(self, corpus_id: str = "NEW_TAB_EN_US", start_date: str=None, user_id=None) -> List[CorpusItem]:
+        if not start_date:
+            start_date = date.today().strftime("%Y-%m-%d")
+
+        request_headers = {
+            "apollographql-client-name": "recommendations-api",
+            "apollographql-client-version": "1"
+        }
+
+        injection_protected_id = str(corpus_id)
+        injection_protected_start_date = str(start_date)
```

Comment:
Converting to string doesn't protect against injection. Try running this:
```suggestion
        start_date = '2022-02-22", arg2: "blah"'
        injection_protected_start_date = str(start_date)
```
Note that the date string contains an unesceaped `"` character. The GraphQL interpreter will interpret that as the end of the string, and treat `arg2` as a second query parameter to `items`. Adding a query parameter is pretty harmless, but maybe a specially crafted string could run a totally different query or mutation.

I'll see if I can get the JSON stringify to work quickly.